full-moon/src/ast/compound.rs

```diff
@@ -0,0 +1,140 @@
+use crate::ast::{Expression, Var};
+use crate::tokenizer::{Symbol, TokenReference};
+use derive_more::Display;
+use full_moon_derive::{Node, Visit};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Display, PartialEq, Eq, Node, Visit)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[non_exhaustive]
+#[allow(missing_docs)]
+#[display("{_0}")]
+/// Compound operators, such as X += Y or X -= Y
+pub enum CompoundOp {
+    PlusEqual(TokenReference),
+    MinusEqual(TokenReference),
+    StarEqual(TokenReference),
+    SlashEqual(TokenReference),
+    CaretEqual(TokenReference),
+
+    // luau sepcific
+    DoubleSlashEqual(TokenReference),
+    PercentEqual(TokenReference),
+    TwoDotsEqual(TokenReference),
+
+    // cfxlua sepcific
+    LeftShift(TokenReference),
+    RightShift(TokenReference),
+    BitwiseAndAssignment(TokenReference),
+    BitwiseOrAssignment(TokenReference),
```

Comment:
To align with the other options. We try to make the names represent the characters rather than the logic, since in future `>>=` could mean something other than right shift operator for a different Lua dialect

```suggestion
    DoubleLessThanEqual(TokenReference),
    DoubleGreaterThanEqual(TokenReference),
    AmpersandEqual(TokenReference),
    PipeEqual(TokenReference),
```